qiskit_nature/second_q/drivers/psi4d/psi4driver.py

```diff
@@ -0,0 +1,247 @@
+#
+# (C) Copyright IBM 2018, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Psi4 Driver """
+
+from __future__ import annotations
+
+import logging
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Any
+
+from qiskit_nature import QiskitNatureError
+from qiskit_nature.exceptions import UnsupportMethodError
+from qiskit_nature.units import DistanceUnit
+from qiskit_nature.second_q.problems import ElectronicBasis, ElectronicStructureProblem
+from qiskit_nature.second_q.formats.molecule_info import MoleculeInfo
+from qiskit_nature.second_q.formats.qcschema import QCSchema
+from qiskit_nature.second_q.formats.qcschema_translator import qcschema_to_problem
+import qiskit_nature.optionals as _optionals
+
+from ..electronic_structure_driver import ElectronicStructureDriver, MethodType, _QCSchemaData
+
+logger = logging.getLogger(__name__)
+
+
+@_optionals.HAS_PSI4.require_in_instance
+class Psi4Driver(ElectronicStructureDriver):
+    """
+    Qiskit Nature driver using the PSI4 program.
+    See http://www.psicode.org/
+    """
+
+    def __init__(
+        self,
+        config: str
+        | list[
+            str
+        ] = "molecule h2 {\n  0 1\n  H  0.0 0.0 0.0\n  H  0.0 0.0 0.735\n  no_com\n  no_reorient\n}\n\n"
+        "set {\n  basis sto-3g\n  scf_type pk\n  reference rhf\n",
+    ) -> None:
+        """
+        Args:
+            config: A molecular configuration conforming to PSI4 format.
+        Raises:
+            QiskitNatureError: Psi4 Driver configuration should be a string or list of strings
+        """
+        super().__init__()
+        if not isinstance(config, str) and not isinstance(config, list):
+            raise QiskitNatureError(
+                f"Psi4 Driver configuration should be a string or list of strings:'{config}'"
+            )
+
+        if isinstance(config, list):
+            config = "\n".join(config)
+
+        self._config = config
+        self._qcschemadata = _QCSchemaData()
+
+    @staticmethod
+    @_optionals.HAS_PSI4.require_in_call
+    def from_molecule(
+        molecule: MoleculeInfo,
+        basis: str = "sto3g",
+        method: MethodType = MethodType.RHF,
+        driver_kwargs: dict[str, Any] | None = None,
+    ) -> "Psi4Driver":
+        """
+        Args:
+            molecule: molecule
+            basis: basis set
+            method: the classical method to run in Psi4
+            driver_kwargs: kwargs to be passed to driver
+        Returns:
+            driver
+        Raises:
+            QiskitNatureError: Unknown unit
```

Comment:
Capitalization inconsistent with other places.

Reply:
This could indeed use some improvements. Same holds for the `PySCFDriver.from_molecule` and `GaussianDriver.from_molecule` methods. We can do these updates as part of our API update sweep of the entire `qiskit_nature.second_q` module later this week :+1: